frog/imports/client/AccountModal/SignUp.js

```diff
@@ -0,0 +1,229 @@
+// @flow
+import React from 'react';
+import {
+  Avatar,
+  Button,
+  TextField,
+  Link,
+  Grid,
+  Container,
+  Typography
+} from '@material-ui/core';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import { Meteor } from 'meteor/meteor';
+import { withStyles } from '@material-ui/styles';
+import {
+  errorBasedOnChars,
+  emailErrors,
+  passwordErrors
+} from './validationHelpers';
+
+type FormError = {
+  displayName: string,
+  email: string,
+  password: string
+};
+
+type SignUpStateT = {
+  displayName: string,
+  email: string,
+  password: string,
+  formErrors: FormError
+};
+
+type SignUpPropsT = {
+  classes: Object,
+  onSignUpSuccess: () => void
+};
+
+const styles = (theme: Object) => ({
+  paper: {
+    marginTop: theme.spacing(1),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(3)
+  },
+  submit: {
+    margin: theme.spacing(2, 0, 2, 0)
+  }
+});
+
+class SignUp extends React.Component<SignUpPropsT, SignUpStateT> {
+  constructor() {
+    super();
+    this.state = {
+      displayName: '',
+      email: '',
+      password: '',
+      formErrors: {
+        displayName: '',
+        email: '',
+        password: ''
+      }
+    };
+  }
+
+  formValid = (formErrors: FormError): boolean => {
+    let valid = true;
+    if (
+      formErrors.email === '' &&
+      formErrors.password === '' &&
+      formErrors.displayName === ''
+    )
+      return valid;
+
+    valid = false;
+    return valid;
+  };
+
+  clearErrors = (): void => {
+    const formErrorsCleared = {
+      displayName: '',
+      email: '',
+      password: ''
+    };
+    this.setState({ formErrors: formErrorsCleared });
+  };
+
+  // Hack to make sure modal doesn't jump on displaying errors
+  displayErrors = (type: string): string => {
+    if (this.state.formErrors[type] === '') return '';
+    else return this.state.formErrors[type];
```

Comment:
This function does not accomplish useful work. If `this.state.formErrors[type] === ''`, then `return this.state.formErrors[type];` is equivalent to `return '';`

Reply:
Thanks! Addressed